backend/src/auth_service/client_api/user_profile.rs

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use phnxtypes::{
+    errors::auth_service::{GetUserProfileError, UpdateUserProfileError},
+    messages::client_as_out::{
+        GetUserProfileParams, GetUserProfileResponse, UpdateUserProfileParamsTbs,
+    },
+};
+
+use crate::auth_service::{AuthService, user_record::UserRecord};
+
+impl AuthService {
+    pub(crate) async fn as_get_user_profile(
+        &self,
+        params: GetUserProfileParams,
+    ) -> Result<GetUserProfileResponse, GetUserProfileError> {
+        let GetUserProfileParams { client_id } = params;
+
+        let user_record = UserRecord::load(&self.db_pool, client_id.user_name())
```

Comment:
Just a note: We could also store the encrypted user profile in S3-like storage. However, the question under which key. This could be the hashed client_id, but then only a single profile per user can be stored. Otherwise, it should be possible to use the encrypted profile key for that. Maybe we could discuss it as a possible future change.

Reply:
I think the plan is to compromise here and store larger things like the profile picture on S3 and then store on the AS DB only a struct containing things like the S3 URL, key, nonce, etc. We can do that when we also do attachments/assets.